drivers/sx127x/sx127x.c

```diff
@@ -0,0 +1,461 @@
+/*
+ * Copyright (C) 2016 Unwired Devices <info@unwds.com>
+ *               2017 Inria Chile
+ *               2017 Inria
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     drivers_sx127x
+ * @{
+ * @file
+ * @brief       Basic functionality of sx127x driver
+ *
+ * @author      Eugene P. <ep@unwds.com>
+ * @author      José Ignacio Alamos <jose.alamos@inria.cl>
+ * @author      Alexandre Abadie <alexandre.abadie@inria.fr>
+ * @}
+ */
+#include <stdbool.h>
+#include <math.h>
+#include <string.h>
+#include <stdlib.h>
+
+#include "xtimer.h"
+#include "thread.h"
+
+#include "periph/gpio.h"
+#include "periph/spi.h"
+
+#include "sx127x.h"
+#include "sx127x_internal.h"
+#include "sx127x_registers.h"
+#include "sx127x_netdev.h"
+
+#define ENABLE_DEBUG (0)
+#include "debug.h"
+
+/* Internal functions */
+static void _init_isrs(sx127x_t *dev);
+static void _init_timers(sx127x_t *dev);
+static int _init_peripherals(sx127x_t *dev);
+static void _on_tx_timeout(void *arg);
+static void _on_rx_timeout(void *arg);
+
+/* SX127X DIO interrupt handlers initialization */
+static void sx127x_on_dio0_isr(void *arg);
+static void sx127x_on_dio1_isr(void *arg);
+static void sx127x_on_dio2_isr(void *arg);
+static void sx127x_on_dio3_isr(void *arg);
+
+
+void sx127x_setup(sx127x_t *dev, const sx127x_params_t *params)
+{
+    netdev_t *netdev = (netdev_t*) dev;
+    netdev->driver = &sx127x_driver;
+    memcpy(&dev->params, params, sizeof(sx127x_params_t));
+}
+
+void sx127x_reset(sx127x_t *dev)
+{
+    /*
+     * This reset scheme complies with 7.2 chapter of the SX1272/1276 datasheet
+     * See http://www.semtech.com/images/datasheet/sx1276.pdf for SX1276
+     * See http://www.semtech.com/images/datasheet/sx1272.pdf for SX1272
+     *
+     * 1. Set NReset pin to LOW for at least 100 us
+     * 2. Set NReset in Hi-Z state
+     * 3. Wait at least 5 milliseconds
+     */
+
+    gpio_init(dev->params.reset_pin, GPIO_OUT);
+
+    /* Set reset pin to 0 */
+    gpio_clear(dev->params.reset_pin);
+
+    /* Wait 1 ms */
+    xtimer_usleep(1000);
+
+    /* Put reset pin in High-Z */
+    gpio_init(dev->params.reset_pin, GPIO_OD);
```

Comment:
after discussing IRL with @haukepetersen, configuring as input seems reasonable. Can you test that it works with your SAML21 ?